extensions/src/ACRE2Core/BabbelEffect.h

```diff
@@ -16,52 +17,50 @@ class CBabbelEffect : public CSoundMonoEffect {
 private:
     Dsp::SimpleFilter<Dsp::Butterworth::LowPass <4>, 1> lpFilter1;
     Dsp::SimpleFilter<Dsp::Butterworth::LowPass <4>, 1> lpFilter2;
-    float offset;
-    float sfreq;
-    float mfreq;
+    float32_t m_offset;
+    float32_t m_sfreq;
+    float32_t m_mfreq;
 public:
     CBabbelEffect() {
-        offset = 128;
-        sfreq = 48000;
-        mfreq = 5200;
-        lpFilter1.setup(4, sfreq, mfreq*0.3);
-        lpFilter2.setup(4, sfreq, mfreq*0.5);
+        m_offset = 128.0f;
+        m_sfreq = 48000.0f;
+        m_mfreq = 5200.0f;
+        lpFilter1.setup(4, (float64_t) m_sfreq, (float64_t) m_mfreq*0.3);
+        lpFilter2.setup(4, (float64_t) m_sfreq, (float64_t) m_mfreq*0.5);
     };
-    void process(short *samples, int sampleCount) {    
-        float *floatPointer[1], *buffer;
-        buffer = new float[sampleCount];
+    void process(int16_t *const samples, const int32_t sampleCount) {
+        float32_t *floatPointer[1], *buffer;
+        buffer = new float32_t[sampleCount];
         floatPointer[0] = buffer;
-        short *tempSamples = new short[sampleCount];
-        memcpy(tempSamples, samples, sizeof(short)*sampleCount);
-        short maxAmp = 1;
-        for (int i = 0; i < sampleCount; ++i) {
-            samples[i] = (short)(((float)tempSamples[i]*1.0f)*cos(((float)i/(sfreq/mfreq))*1.4*M_PI));
+        int16_t *tempSamples = new int16_t[sampleCount];
```

Comment:
use std::vector with resize for a temp array. That way you don't have to worry about cleanup.